demo-extended/src/main/java/com/novoda/downloadmanager/demo/extended/MainActivity.java

```diff
@@ -33,6 +34,19 @@ protected void onCreate(Bundle savedInstanceState) {
         com.novoda.notils.logger.simple.Log.setShowLogs(true);
         setContentView(R.layout.activity_main);
         listView = (ListView) findViewById(R.id.main_downloads_list);
+        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+                DownloadAdapter adapter = (DownloadAdapter) parent.getAdapter();
+                Download item = adapter.getItem(position);
+                long batchId = item.getBatchId();
+                if (item.isPaused()) {
```

Comment:
the idea will be to have a menu this menu shows all the features as a list, this has two benefits: 1) in the code you can easily see how each function is implemented 2) in the demo you can understand how it works.   Granted a few can go together but the `MainActivity` is no longer easily understandable for someone coming fresh to the project and thinking ok how do I do X